listenbrainz/webserver/views/atom.py

```diff
@@ -0,0 +1,242 @@
+from datetime import datetime, timedelta, date, timezone
+from feedgen.feed import FeedGenerator
+from flask import Blueprint, Response, current_app, request, render_template
+from listenbrainz.webserver.decorators import crossdomain, api_listenstore_needed
+from brainzutils.ratelimit import ratelimit
+import listenbrainz.db.user as db_user
+from listenbrainz.webserver import db_conn, timescale_connection
+from listenbrainz.webserver.views.api_tools import _parse_bool_arg, _parse_int_arg
+from listenbrainz.webserver.views.explore_api import (
+    DEFAULT_NUMBER_OF_FRESH_RELEASE_DAYS,
+    MAX_NUMBER_OF_FRESH_RELEASE_DAYS,
+)
+from listenbrainz.webserver import db_conn, ts_conn
+from listenbrainz.db.fresh_releases import get_sitewide_fresh_releases
+from listenbrainz.db.fresh_releases import get_fresh_releases as db_get_fresh_releases
+
+atom_bp = Blueprint("atom", __name__)
+
+
+@atom_bp.route("/user/<user_name>/listens", methods=["GET"])
+@crossdomain
+@ratelimit()
+@api_listenstore_needed
+def get_listens(user_name):
+    """
+    Get listens feed for a user.
+
+    :param interval: The time interval in minutes from current time to fetch listens for. For example, if interval=60, listens from the last hour will be fetched. Default is 60.
```

Comment:
We will need to set an upper limit on this call as well. Giving an argument of 10000000 would take a looong time to execute and would result in a DoS attack on our database server. What would be a good upper limit? A week?

Reply:
Thanks for point it out. A little math: a song ~ 3 minutes; listening to music nonstop for 8 hours a day => 160 songs. A week gives us 1120 songs. Similar to 
https://github.com/metabrainz/listenbrainz-server/blob/f82daad555a5e26af02bb0b73a7c64cc7767d35f/listenbrainz/webserver/views/api_tools.py#L44
So I think we can handle a week. And for feed reading, a week is more than enough. So I think setting the upper limit to a week is good!